core/embed/bootloader/main.c

```diff
@@ -413,6 +394,7 @@ int bootloader_main(void) {
   volatile secbool header_present = secfalse;
   volatile secbool firmware_present = secfalse;
   volatile secbool firmware_present_backup = secfalse;
+  volatile secbool auto_upgrade = secfalse;
```

Comment:
I think we should make this with the `_backup` with `ensure` version to prevent glitching. Not entirely sure if it's 100% required, but making upgrade harder to glitch is better. OTOH user could get to upgrade by pressing buttons, but maybe there is some weird combination of somewhat-glitched state that could make it useful.